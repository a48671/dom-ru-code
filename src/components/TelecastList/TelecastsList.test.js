import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import TelecasrsList from './index';

const info1 = [{ title: 'Title', time: '12:00', type: 'isOver' }];
const info2 = [
  { title: 'Title', time: '12:00', type: 'isOver' },
  { title: 'Title2', time: '13:00', type: 'isFuture' },
];

describe('TelecasrsList', () => {
  it('render correctly component', () => {
    const TelecasrsListComponent = renderer
      .create(<TelecasrsList info={info1} />)
      .toJSON();
    expect(TelecasrsListComponent).toMatchSnapshot();
  });

  it('should be the right amount TelecastsItems', () => {
    const container = render(<TelecasrsList info={info2} />);
    expect(container.find('.telecast-item').length).toEqual(info2.length);
  });
});
