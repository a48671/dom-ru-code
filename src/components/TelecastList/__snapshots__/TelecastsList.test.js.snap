// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelecasrsList render correctly component 1`] = `
<div
  className="sc-bdVaJa telecast-list eVwNQw"
>
  <div
    className="sc-bwzfXH telecast-item ccuYGw"
    type="isOver"
  >
    <div
      className="sc-htpNat telecast-time zkILo"
    >
      12:00
    </div>
    <h1
      className="sc-bxivhb head__title gsnRvY"
    >
      Title
    </h1>
  </div>
</div>
`;
